### PR TITLE
docs: Migrate separation logic theory reference.

### PR DESCRIPTION
**Pull request by aniemetz**
This migrates page https://cvc4.github.io/separation-logic.

### CODE REVIEW

**Comment by aniemetz** [comment]
https://cvc5.github.io/docs-ci/docs-pr6702/

**Comment by ajreynol** [comment]
ALL_SUPPORTED should change to ALL right?

File: docs/theories/separation-logic.rst [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+
+Syntax
+------
+
+Separation logic in cvc5 requires the ``QF_ALL`` logic.
+
+The syntax for the operators of separation logic is summarized in the following
+table.
+For the C++ API examples in this table, we assume that we have created
+a `cvc5::api::Solver solver` object.
```
Comment (nafur) [inline docs/theories/separation-logic.rst:72]:
```suggestion
a :cpp:class:`cvc5::api::Solver` object.
```

File: docs/theories/separation-logic.rst [hunk 2]
```diff
@@ -0,0 +1,154 @@
+Theory Reference: Separation Logic
+==================================
+
+cvc5 supports a syntax for separation logic as an extension of the SMT-LIB 2
+language.
+
+Signature
+---------
+
+Given a (decidable) base theory ``T``, cvc5 has a
```
Comment (nafur) [inline docs/theories/separation-logic.rst:10]:
```suggestion
Given a (decidable) base theory :math:`T`, cvc5 has a
```

File: docs/theories/separation-logic.rst [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+`decision procedure <https://cvc4.github.io/publications/2016/RIS+16.pdf>`__
+for quantifier-free :math:`SL(T)_{Loc,Data}` formulas, where ``Loc`` and
+``Data`` are any sort belonging to ``T``.
```
Comment (nafur) [inline docs/theories/separation-logic.rst:13]:
```suggestion
``Data`` are any sort belonging to :math:`T`.
```

File: docs/theories/separation-logic.rst [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+
+A :math:`SL(T)_{Loc,Data}` formula is one from the following grammar:
+
+.. code::
+
+  F : L | (emp t u) | (pto t u) | (sep F1 ... Fn) | (wand F1 F2) | ~F1 | F1 op ... op Fn
+
+where ``op`` is any classical Boolean connective, ``t`` and ``u`` are terms
+built from symbols in the signature of ``T`` of sort ``Loc`` and ``Data``
+respectively, and ``L`` is a ``T``-literal.
```
Comment (nafur) [inline docs/theories/separation-logic.rst:23]:
```suggestion
built from symbols in the signature of :math:`T` of sort ``Loc`` and ``Data``
respectively, and ``L`` is a :math:`T`-literal.
```

Comment (nafur) [reply docs/theories/separation-logic.rst:23]: I guess also for Loc and Data?

**Review by nafur — COMMENTED** [review]
Some minor improvements.

**Review by nafur — APPROVED** [review]
LGTM.

**Pull request merged**